packages/esm-patient-chart-app/src/banner/ui-components/overflow-menu.component.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { useState, useCallback, useEffect, useRef } from 'react';
+
+interface CustomOverflowMenuComponentProps {
+  menuTitle: React.ReactNode;
+  dropDownMenu: boolean;
+  children: React.ReactNode;
+}
+
+const CustomOverflowMenuComponent: React.FC<CustomOverflowMenuComponentProps> = ({
+  dropDownMenu,
+  menuTitle,
+  children,
+}) => {
+  const [showMenu, setShowMenu] = useState(false);
+  const wrapperRef = useRef(null);
+
+  useEffect(() => {
+    if (!dropDownMenu) {
+      setShowMenu((state) => state);
+    }
+
+    setShowMenu(() => false);
+  }, [dropDownMenu]);
+  const toggleShowMenu = useCallback(() => setShowMenu((state) => !state), []);
+
+  useEffect(() => {
+    /**
+     * Toggle showMenu if clicked on outside of element
+     */
+    function handleClickOutside(event: MouseEvent) {
+      if (wrapperRef.current && !wrapperRef.current.contains(event.target)) {
+        setShowMenu(false);
+      }
+    }
+
+    // Bind the event listener
+    document.addEventListener('mousedown', handleClickOutside);
+    return () => {
+      // Unbind the event listener on clean up
+      document.removeEventListener('mousedown', handleClickOutside);
+    };
+  }, [wrapperRef]);
+
+  return (
+    <div
+      data-overflow-menu
+      className="cds--overflow-menu"
```

Comment:
Using classnames from carbon is a good indication that we should probably be using their actual component. In this case https://react.carbondesignsystem.com/?path=/story/components-overflowmenu--default . @hadijahkyampeire are you able to clean this one? and for the child components as well. This seems like a lot of customization for what should be an out-of-the-box component from Carbon.

Reply:
This is somehow tricky because the carbon design component will not bring out the design as it is desired here on its own, we might need to combine it with custom `JSX` and that might require some more time to get it to work.